sdk/helper/ldaputil/client.go

```diff
@@ -489,5 +489,12 @@ func getTLSConfig(cfg *ConfigEntry, host string) (*tls.Config, error) {
 		}
 		tlsConfig.RootCAs = caPool
 	}
+	if cfg.ClientTLSCert != "" && cfg.ClientTLSKey != "" {
```

Comment:
Would you mind to add an extra check that checks if only one option has been set? For example: 
https://github.com/hashicorp/vault/blob/f6547fa8e820b6ebbfa15018477a138b38707d91/sdk/helper/tlsutil/tlsutil.go#L168-L170

Reply:
Could you explain better the extra check? Normally we want both the certificate and the key. Is there any case where we want one and not the other?

Reply:
The code already checks if _both_ have been set, I think what's needed is an `else` branch that bubbles the error up to the client if one has been set and not the other.

Reply:
Added check and proper error message